packages/components/modules/messages/CreateGroup/index.tsx

```diff
@@ -0,0 +1,356 @@
+import {
+  CheckMarkIcon,
+  CircledAvatar,
+  CloseIcon,
+  Searchbar as DefaultSearchbar,
+  FileUploadButton,
+  IconButton,
+  LoadingState,
+  TextField,
+} from '@baseapp-frontend/design-system'
+import { filterDirtyValues } from '@baseapp-frontend/utils'
+
+import { zodResolver } from '@hookform/resolvers/zod'
+import { LoadingButton } from '@mui/lab'
+import { Box, Typography, useTheme } from '@mui/material'
+import { useForm } from 'react-hook-form'
+import { Virtuoso } from 'react-virtuoso'
+import { ConnectionHandler } from 'relay-runtime'
+
+import { useAllProfilesList } from '../../profiles/graphql/queries/AllProfilesList'
+import { ProfileNode } from '../CreateChatRoomList/types'
+import EmptyProfilesListState from '../EmptyProfilesListState'
+import SearchNotFoundState from '../SearchNotFoundState'
+import { useChatRoom } from '../context'
+import { useCreateChatRoomMutation } from '../graphql/mutations/CreateChatRoom'
+import DefaultProfileCard from './ProfileCard'
+import {
+  DEFAULT_FORM_VALIDATION,
+  DEFAULT_FORM_VALUES,
+  DEFAULT_IMAGE_FORMATS,
+  DEFAULT_IMAGE_MAX_SIZE,
+  FORM_VALUE,
+} from './constants'
+import { HeaderContainer, SearchbarContainer, UploadProfileContainer } from './styled'
+import { CreateGroupProps } from './types'
+import { getImageUrl } from './utils'
+
+const CreateGroup: FC<CreateGroupProps> = ({
+  allProfilesRef,
+  // TODO: Check if this is the correct way to get the remotePatternsHostName
+  remotePatternsHostName = process.env.NEXT_PUBLIC_REMOTE_PATTERNS_HOSTNAME || '', // To get this working in staging/prod, change NEXT_PUBLIC_REMOTE_PATTERNS_HOSTNAME to the media host being used (e.g.: digitalocean, aws, etc)
+  ProfileCard = DefaultProfileCard,
+  ProfileCardProps = {},
+  Searchbar = DefaultSearchbar,
+  SearchbarProps = {},
+  VirtuosoProps = {},
+  setIsInGroupChatCreation,
+  setIsInExistingChatRoomsView,
+}) => {
+  const theme = useTheme()
+  const {
+    data: { allProfiles },
+    loadNext,
+    isLoadingNext,
+    hasNext,
+    refetch,
+  } = useAllProfilesList(allProfilesRef)
+  const [isPending, startTransition] = useTransition()
+  const {
+    control: searchControl,
+    reset: searchReset,
+    watch: searchWatch,
+  } = useForm({ defaultValues: { search: '' } })
+
+  const {
+    control,
+    setValue,
+    watch,
+    getFieldState,
+    clearErrors,
+    reset,
+    handleSubmit,
+    formState: { isValid, isDirty, dirtyFields },
+  } = useForm({
+    defaultValues: DEFAULT_FORM_VALUES,
+    resolver: zodResolver(DEFAULT_FORM_VALIDATION),
+    mode: 'onBlur',
+  })
+
+  const { currentProfile } = useCurrentProfile()
+
+  const [commit, isMutationInFlight] = useCreateChatRoomMutation()
+  const { setChatRoom } = useChatRoom()
+
+  const onSubmit = handleSubmit((data: any) => {
+    const dirtyValues = filterDirtyValues({ values: data, dirtyFields })
+    const { title, participants, image } = data
+    const participantsIds = participants.map((member: any) => member.id)
```

Comment:
we could better type that
```suggestion
  const onSubmit = handleSubmit((data: CreatGroupUpload) => {
    const dirtyValues = filterDirtyValues({ values: data, dirtyFields })
    const { title, participants, image } = data
    const participantsIds = participants?.map((member) => member!.id) || []
```